File: MIDIJava/src/io/github/seanboyy/Midi/File.java
package io.github.seanboyy.Midi;

import java.io.DataInputStream;
import java.io.BufferedInputStream;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.DataOutputStream;
import java.io.BufferedOutputStream;
import java.io.FileOutputStream;

class File{
	static Midi read(String filename) throws IOException{
		java.io.File file = new java.io.File(filename);
		DataInputStream dis = new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
		byte[] fileString = new byte[(int)file.length()];
		int count = dis.read(fileString);
		assert(count == (int)file.length());
		Midi midi = new Midi();
		midi.fromBitString(fileString, (int)file.length());
		dis.close();
		return midi;
	}
	
	static void write(String filename, Midi midi) throws IOException{
		java.io.File file = new java.io.File(filename);
		DataOutputStream dos = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
		dos.write(midi.toBitString(), 0, midi.getLength());
		dos.flush();
		dos.close();
	}
}

File: MIDIC#/MIDILibrary/MIDILibrary/File.cs
using System.IO;

namespace MIDILibrary
{
    public class File
    {
        public static Midi Read(string filename)
        {
            var fs = new FileStream(filename, FileMode.Open);
            var len = (int)fs.Length;
            var bits = new byte[len];
            fs.Read(bits, 0, len);
            var midi = new Midi();
            midi.FromBitString(bits, len);
            fs.Close();
            return midi;
        }

        public static void Write(string filename, Midi midi)
        {
            var fs = new FileStream(filename, FileMode.OpenOrCreate);
            fs.Write(midi.ToBitString(), 0, (int)midi.GetLength());
            fs.Flush();
            fs.Close();
        }
    }
}
